### ISSUE TITLE
Add Facet/Filter Without Scrolling to Column

### ISSUE THREAD

**Issue by grahamjevon** [OpenRefine/OpenRefine#3254]
I find it frustrating and cumbersome flitting back and forth between columns and facets. It would be good if the permanent facets bar enabled more actions. There are 2 sides to this problem:

**1. Adding new Facets/Filters**
I have many columns, I mostly know the column names, but I don't always know where the column is located in the dataset. To find a column so that I can add a facet or filter, I use ctrl+F, enter part of the column name, and then scroll along until I spot the highlighted section. This can sometimes feel cumbersome. It also means I lose my place if I want the screen to display a different portion of the data.

**2. Editing columns with a facet open**

When I am working with a facet, I might then want to then edit cells in that column or add column based on the column being faceted/filtered. Again, to do this I need to find that column (as above).

### Proposed solution

**1. Adding new Facets/Filters**
It would be easier if the permanent facet bar had an "Add Facet" and "Add Filter" option. Clicking this would open the normal facet/filter menu options available via the column header. Either before or after making this selection, perhaps it could open a vertical list of column names for me to select from. Or, better still, enable me to start entering the column name so that it narrows down the column options.

**2. Editing columns with a facet open**

It would be easier if the facet/filter included an option to open the menu options that are currently only accessible by clicking on the arrow located on the column header.

### Additional context

![OpenRefineFacetsRequest](https://user-images.githubusercontent.com/51366817/95465648-51036c80-0973-11eb-96e4-3f7055a28889.PNG)

**Comment by thadguidry** [comment]
For # 1 , So clicking on the + plus icon then a quick search input box appears to begin to type in a column name that shows a drop down list, the user clicks on 1 column name in the drop down list, or does he simply hover the mouse on the column name for .5 secs and then it hovers a submenu box to show only the Facet submenu and Text Filter (not showing other submenus click Edit Cells, Edit Column, since the idea is that they could use # 2 idea to quickly navigate for fuller options to perform).

For # 2, perhaps a better idea is to change the current title of the facet
Make it color blue #1d59db

**CSS:**
```css
color: #1d59db;
```
**HTML:**
 `<span bind="titleSpan">`

 to

     `<a class="facet-name" href="javascript:{} bind="titleSpan">`

to make it a clickable element that once the user clicks, an event fires to scroll the data grid horizontally to the column position for you, and minimally trying to present it on the data grid as the 2nd position in the view (knowing that good parentage details might want to be seen in the 1st position of the view to compare with the column.  Rather than simply scrolling it into the 1st position and you would have to move the grid scroll over a bit to see some parentage.  In my example, once the `xxxx` element is clicked, it would scroll the grid to the right for you until it either puts the `xxxx` column into the 2nd position or the horizontal scroll hits the end of the view while attempting to put it into the 2nd position of the view.

![image](https://user-images.githubusercontent.com/986438/95477997-96617300-094e-11eb-80b6-5eaa5866e6e1.png)

![image](https://user-images.githubusercontent.com/986438/95478667-71b9cb00-094f-11eb-92c3-8131c5c5133c.png)

**Comment by grahamjevon** [comment]
For # 1 click versus .5 second hover I'm probably ok with either.

Regarding menu options for #1, I was initially thinking just facet/filter options. But on reflection, I would probably favour all menu options. This feature would then somewhat solve #2. And it would have the advantage of enabling quick access to edit cells on a column for which a facet/filter is not currently present. However, I might still favour #2 as well because if I have a facet/filter present, I might quickly get frustrated with having to keep going through the #1 process to edit cells when a menu option on the facet would be more efficient.

For #2 idea of scrolling to the column, I am undecided as to whether not that is preferable (eg. maybe I would prefer not to lose my current position in the dataset). I'll think further on this. The scrolling idea might be more attractive if #1 includes all menu options. That way you could edit cells using #1 if you don't want to scroll, but could use #2 if you do want to scroll.

**Comment by thadguidry** [comment]
@grahamjevon The data grid would only scroll horizontally once you clicked on the blue title of the facet, otherwise nothing moves on the data grid.

**Comment by tfmorris** [comment]
These sound like two independent feature requests. Is there a reason that they need to be treated together?

**Comment by grahamjevon** [comment]
I did indeed question whether to add them separately at the outset. The commonality was that these are both means of accessing the column actions menu from the permanent facets bar to avoid the problem of flitting back and forth. I wonder if there are now 3 feature requests here:
1. Access column actions menu for any column via + sign (type column name to help identify column)
2. Access column actions menu from facet/filter
3. Scroll to column by clicking on column name in facet/filter

I like Thad's idea (3) of being able to auto scroll to a column. I think I there occasions where I would use that. If I'm not mistaken, @thadguidry is suggesting 3 instead of 2. For me, personally, I think there are also occasions when I would use 2 rather than 3 - e.g. I may want to edit a faceted column not visible on screen, without losing my grid position (but maybe 2 would be less necessary if 1 and 3 were available)

**Comment by OmarMorales71** [comment]
@thadguidry Can I work on this issue?

**Comment by thadguidry** [comment]
@OmarMorales71 YES!  However, a warning on this particular issue.  This will require some dedication... There will be several feedback rounds as you make a draft PR request as we review your work while you progress and we test and you make changes and updates...back and forth...until everyone is happy.  This specific issue might take weeks and likely over a month with many hours of your volunteer time.

If you still feel confident and want to progress, then by all means, we will gladly accept your volunteer enthusiasm on this issue to give our users something perfect!

Let us know!
